From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Narayan Reddy <narayanr@nvidia.com>
Date: Mon, 23 Aug 2021 01:53:58 +0530
Subject: [PATCH 027/243] NVIDIA: SAUCE: net: phy: add 10G, 5G for fixed link

Issue: currently there is no support for
10G, 5G fixed link speed

Fix: Add 10G and 5G speed entries for the
fixed link case

Bug 3978991

Change-Id: I432cc5f498594bdbafde8273163d353a738c999d
Signed-off-by: Narayan Reddy <narayanr@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/linux-5.10/+/2581148
Reviewed-on: https://git-master.nvidia.com/r/c/linux-stable/+/2817669
Signed-off-by: Revanth Kumar Uppala <ruppala@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/linux-stable/+/2872938
Reviewed-by: Laxman Dewangan <ldewangan@nvidia.com>

diff --git a/drivers/net/phy/swphy.c b/drivers/net/phy/swphy.c
index 59f1ba4d49bc..6e946285fa0f 100644
--- a/drivers/net/phy/swphy.c
+++ b/drivers/net/phy/swphy.c
@@ -8,6 +8,7 @@
  *         Anton Vorontsov <avorontsov@ru.mvista.com>
  *
  * Copyright (c) 2006-2007 MontaVista Software, Inc.
+ * Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
  */
 #include <linux/export.h>
 #include <linux/mii.h>
@@ -29,6 +30,8 @@ enum {
 	SWMII_SPEED_10 = 0,
 	SWMII_SPEED_100,
 	SWMII_SPEED_1000,
+	SWMII_SPEED_5000,
+	SWMII_SPEED_10000,
 	SWMII_DUPLEX_HALF = 0,
 	SWMII_DUPLEX_FULL,
 };
@@ -71,6 +74,10 @@ static const struct swmii_regs duplex[] = {
 static int swphy_decode_speed(int speed)
 {
 	switch (speed) {
+	case 10000:
+		return SWMII_SPEED_10000;
+	case 5000:
+		return SWMII_SPEED_5000;
 	case 1000:
 		return SWMII_SPEED_1000;
 	case 100:
-- 
2.43.0

